File: src/models/output/binary.jl
"""
Output module
Pengfei Cheng

Create the DataFrame for the binary variables.
"""

function gen_binary_df(m)

    binary_vars = ["y", "z0", "z"]

    # add columns
    l0 = []
    for v in binary_vars
        local l = [v => Float64[]]
        # extend array l0
        l0 = vcat(l0, l)
    end

    df_binary = DataFrame(l0)
    # add the values to the DataFrame

    for i in set_hour
        local l0 = []
        for v in binary_vars
            s = Symbol(v)
            local l = [value(m[s][i])]
            l0 = vcat(l0, l)
        end
        push!(df_binary, l0)
    end

    return df_binary
end

File: src/pyomo_model/output/binary.py
"""
Output module
Pengfei Cheng

Create the DataFrame for the binary variables.
"""

import pandas as pd
from pyomo.environ import value
from src.pyomo_model.params import *


def gen_binary_df(m):

    dfs = {}

    vars = [
        "y", "z", "z0",
    ]

    for s in set_scenario:
        dict_list = []
        for var_name in vars:
            var = getattr(m, var_name)
            _tmp_dict = {i: value(var[i, _s]) for i, _s in var if _s == s}
            dict_list.append(_tmp_dict)

        df = pd.DataFrame(dict_list).T
        df.columns = vars

        dfs[s] = df

    return dfs